### PR TITLE
Update for PHP 7.2, 7.3 and 7.4

### PR DESCRIPTION
**Pull request by kenjis**
This PR needs:
- [x] updated Aura.Project_Kernel https://github.com/auraphp/Aura.Project_Kernel/pull/15
- [ ] updated Aura.Cli https://github.com/auraphp/Aura.Cli/pull/65
- [ ] Aura.Project_Kernel 4.0.0 release
- [ ] Aura.Cli 4.0.0 release

### CODE REVIEW

File: config/Common.php [hunk 1]
```diff
@@ -35,12 +35,12 @@ public function define(Container $di)
...
     }
 
-    public function modify(Container $di)
+    public function modify(Container $di): void
     {
         $dispatcher = $di->get('aura/cli-kernel:dispatcher');
         $dispatcher->setObject(
             'help',
-            $di->lazyNew('Aura\Cli_Kernel\HelpCommand')
+            $di->newInstance('Aura\Cli_Kernel\HelpCommand')
```
Comment (kenjis) [inline config/Common.php:43]: I don't know why, but when `$di->lazyNew()`, tests fail.

```sh-session
$ vendor/bin/phpunit 
PHPUnit 7.5.20 by Sebastian Bergmann and contributors.

.F..FEEEE                                                           9 / 9 (100%)

Time: 216 ms, Memory: 6.00 MB

There were 4 errors:

1) Aura\Cli_Kernel\CliKernelTest::testHelpCommand
TypeError: Argument 1 passed to Aura\Cli_Kernel\HelpCommand::__construct() must be an instance of Aura\Cli\Stdio, string given

/Users/kenji/work/aura/Aura.Cli_Kernel/src/HelpCommand.php:62
/Users/kenji/work/aura/Aura.Cli_Kernel/vendor/aura/di/src/Resolver/Blueprint.php:99
/Users/kenji/work/aura/Aura.Cli_Kernel/vendor/aura/di/src/Resolver/Resolver.php:140
/Users/kenji/work/aura/Aura.Cli_Kernel/vendor/aura/di/src/Injection/Factory.php:97
/Users/kenji/work/aura/Aura.Cli_Kernel/vendor/aura/dispatcher/src/InvokeMethodTrait.php:94
/Users/kenji/work/aura/Aura.Cli_Kernel/vendor/aura/dispatcher/src/Dispatcher.php:122
/Users/kenji/work/aura/Aura.Cli_Kernel/vendor/aura/dispatcher/src/Dispatcher.php:95
/Users/kenji/work/aura/Aura.Cli_Kernel/src/CliKernel.php:196
/Users/kenji/work/aura/Aura.Cli_Kernel/src/CliKernel.php:144
/Users/kenji/work/aura/Aura.Cli_Kernel/tests/CliKernelTest.php:25
/Users/kenji/work/aura/Aura.Cli_Kernel/tests/CliKernelTest.php:93

2) Aura\Cli_Kernel\CliKernelTest::testHelpCommandUnvailable
TypeError: Argument 1 passed to Aura\Cli_Kernel\HelpCommand::__construct() must be an instance of Aura\Cli\Stdio, string given

/Users/kenji/work/aura/Aura.Cli_Kernel/src/HelpCommand.php:62
/Users/kenji/work/aura/Aura.Cli_Kernel/vendor/aura/di/src/Resolver/Blueprint.php:99
/Users/kenji/work/aura/Aura.Cli_Kernel/vendor/aura/di/src/Resolver/Resolver.php:140
/Users/kenji/work/aura/Aura.Cli_Kernel/vendor/aura/di/src/Injection/Factory.php:97
/Users/kenji/work/aura/Aura.Cli_Kernel/vendor/aura/dispatcher/src/InvokeMethodTrait.php:94
/Users/kenji/work/aura/Aura.Cli_Kernel/vendor/aura/dispatcher/src/Dispatcher.php:122
/Users/kenji/work/aura/Aura.Cli_Kernel/vendor/aura/dispatcher/src/Dispatcher.php:95
/Users/kenji/work/aura/Aura.Cli_Kernel/src/CliKernel.php:196
/Users/kenji/work/aura/Aura.Cli_Kernel/src/CliKernel.php:144
/Users/kenji/work/aura/Aura.Cli_Kernel/tests/CliKernelTest.php:25
/Users/kenji/work/aura/Aura.Cli_Kernel/tests/CliKernelTest.php:112

3) Aura\Cli_Kernel\CliKernelTest::testHelpCommandHelpUnvailable
TypeError: Argument 1 passed to Aura\Cli_Kernel\HelpCommand::__construct() must be an instance of Aura\Cli\Stdio, string given

/Users/kenji/work/aura/Aura.Cli_Kernel/src/HelpCommand.php:62
/Users/kenji/work/aura/Aura.Cli_Kernel/vendor/aura/di/src/Resolver/Blueprint.php:99
/Users/kenji/work/aura/Aura.Cli_Kernel/vendor/aura/di/src/Resolver/Resolver.php:140
/Users/kenji/work/aura/Aura.Cli_Kernel/vendor/aura/di/src/Injection/Factory.php:97
/Users/kenji/work/aura/Aura.Cli_Kernel/vendor/aura/dispatcher/src/InvokeMethodTrait.php:94
/Users/kenji/work/aura/Aura.Cli_Kernel/vendor/aura/dispatcher/src/Dispatcher.php:122
/Users/kenji/work/aura/Aura.Cli_Kernel/vendor/aura/dispatcher/src/Dispatcher.php:95
/Users/kenji/work/aura/Aura.Cli_Kernel/src/CliKernel.php:196
/Users/kenji/work/aura/Aura.Cli_Kernel/src/CliKernel.php:144
/Users/kenji/work/aura/Aura.Cli_Kernel/tests/CliKernelTest.php:25
/Users/kenji/work/aura/Aura.Cli_Kernel/tests/CliKernelTest.php:124

4) Aura\Cli_Kernel\CliKernelTest::testHelpHelp
TypeError: Argument 1 passed to Aura\Cli_Kernel\HelpCommand::__construct() must be an instance of Aura\Cli\Stdio, string given

/Users/kenji/work/aura/Aura.Cli_Kernel/src/HelpCommand.php:62
/Users/kenji/work/aura/Aura.Cli_Kernel/vendor/aura/di/src/Resolver/Blueprint.php:99
/Users/kenji/work/aura/Aura.Cli_Kernel/vendor/aura/di/src/Resolver/Resolver.php:140
/Users/kenji/work/aura/Aura.Cli_Kernel/vendor/aura/di/src/Injection/Factory.php:97
/Users/kenji/work/aura/Aura.Cli_Kernel/vendor/aura/dispatcher/src/InvokeMethodTrait.php:94
/Users/kenji/work/aura/Aura.Cli_Kernel/vendor/aura/dispatcher/src/Dispatcher.php:122
/Users/kenji/work/aura/Aura.Cli_Kernel/vendor/aura/dispatcher/src/Dispatcher.php:95
/Users/kenji/work/aura/Aura.Cli_Kernel/src/CliKernel.php:196
/Users/kenji/work/aura/Aura.Cli_Kernel/src/CliKernel.php:144
/Users/kenji/work/aura/Aura.Cli_Kernel/tests/CliKernelTest.php:25
/Users/kenji/work/aura/Aura.Cli_Kernel/tests/CliKernelTest.php:136

--

There were 2 failures:

1) Aura\Cli_Kernel\CliKernelTest::testNoCommandSpecified
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-''
+'Aura\Dispatcher\Exception\ParamNotSpecified: Aura\Di\Injection\LazyNew::__invoke(0 : $params)\n
+'

/Users/kenji/work/aura/Aura.Cli_Kernel/tests/CliKernelTest.php:167
/Users/kenji/work/aura/Aura.Cli_Kernel/tests/CliKernelTest.php:51

2) Aura\Cli_Kernel\CliKernelTest::testHelp
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-''
+'Aura\Dispatcher\Exception\ParamNotSpecified: Aura\Di\Injection\LazyNew::__invoke(0 : $params)\n
+'

/Users/kenji/work/aura/Aura.Cli_Kernel/tests/CliKernelTest.php:167
/Users/kenji/work/aura/Aura.Cli_Kernel/tests/CliKernelTest.php:86

ERRORS!
Tests: 9, Assertions: 11, Errors: 4, Failures: 2.
kenji:~/work/aura/Aura.Cli_Kernel (3.x *)$ vendor/bin/phpunit 
PHPUnit 7.5.20 by Sebastian Bergmann and contributors.

.........                                                           9 / 9 (100%)

Time: 119 ms, Memory: 6.00 MB

OK (9 tests, 26 assertions)
```

**Comment by koriym** [comment]
@kenjis Aura has a policy of not accepting PRs that drop support for old PHP. You will need to upgrade to a major version.

See https://github.com/auraphp/Aura.Sql/pull/188.

**Comment by kenjis** [comment]
@koriym Thanks!
There is no branch `3.x`. So I can't send this PR to it.

**Comment by koriym** [comment]
@pmjones What should we do?

**Comment by pmjones** [comment]
Hi @koriym and @kenjis,

I have just created 3.x and 4.x branches for this project.

In the spirit of delegation, I have invited @koriym to become an admin on Aura. (@harikt I don't imagine you have any objections?) Thereafter, @koriym should be able to create the appropriate branches in the future -- let me know if you cannot!

@kenjis I have invited you to the Code Team on Aura; this should give you some privileges here as well.

Some notes:

- The Aura 3.x series was for PHP 5.6 and later.
- The Aura 4.x series is for PHP 7.2 and later.
- I imagine a 5.x series, if you want to start one, should be for 8.0 (or perhaps 8.1?) and later.

Let me know if this begins to help.

**Comment by pmjones** [comment]
Some further comments:

Back in 2015, when the 3.x series started, I did not envision continuing the framework portions of Aura. I expect that if you want to do so here, you will have a heavy load to carry; you may need to update other packages, such as Aura.Cli_Project and Aura.Project_Kernel.

It might be better overall to fork the kernel and project packages under new repositories of your own making. But, if you want to carry that load under Aura, given your (@koriym and @kenjis) long-time standing in the community here, I am happy to help make it possible.

Again, I hope that helps -- please let me know if it does not.

**Comment by koriym** [comment]
@pmjones Thank you for inviting me.

I personally don't plan to use it as a framework either, but if @kenjis wants to keep it under Aura, I can help with that.

@kenjis Let me know when you need 5.x. We'll help you.

**Comment by harikt** [comment]
Oh I am late to the party. Sorry about that. 

No objections adding @koriym or @kenjis . I am carried away with some personal stuffs ie why not able to be part of the community as I wished to.

Thanks for your understanding. Feel free to ping me if you need any help.

**Comment by kenjis** [comment]
@pmjones @harikt @koriym 
Thank you for your replies and kind help.

I would like to make Aura.Cli work on PHP 7.x (and 8).
So I hope this PR against 4.x branch will be merged.

Before merging this PR, updating Aura.Project kernel and Aura.Cli is needed.
So I want to handle this PR first: https://github.com/auraphp/Aura.Project_Kernel/pull/15

**Comment by kenjis** [comment]
@pmjones @harikt @koriym
See https://travis-ci.org/github/auraphp/Aura.Cli_Kernel/builds/763375695
> Since June 15th, 2021, the building on travis-ci.org is ceased. Please use travis-ci.com from now on. 

We can't use travis-ci.org any more. Can we move to  travis-ci.com?
https://docs.travis-ci.com/user/migrate/open-source-repository-migration#migrating-a-repository

**Comment by harikt** [comment]
Hey @kenjis , I tried to migrate, but even after adding the organization, the repos are not shown. Not sure if the owner ( in this case @pmjones ) should add anything. Another way is we integrate the github actions and discard the travis.

**Comment by kenjis** [comment]
@harikt Thank you for your try.
I don't know only owners can migrate or not.
And I'm not sure it might help you, but there is my memo when I got migrated:
https://translate.google.com/translate?sl=ja&tl=en&u=http://blog.a-way-out.net/blog/2021/07/03/migrate-to-travis-ci-com/

**Comment by harikt** [comment]
We don't need to worry about travis now. Github actions is working. So you can merge or ping if you need a review.

**Comment by kenjis** [comment]
Go to #13

**Pull request closed without merging**